### PR TITLE
refactor: Updated instrumentation core - ioredis

### LINKED ISSUE #2022: Update agent to explicitly instantiate specs instead of passing objects to shim methods
### Description
In #2004 work was done to make it easier to understand which specs are needed for given instrumentation.  Any place where `shim.<method>` is used where it expects to return a spec, needs to be updated to explicitly instantiate the proper spec type.  [See](https://github.com/newrelic/node-newrelic/blob/main/lib/instrumentation/openai.js#L269) for an example of this.  Once all instrumented is updated, the various shim helper methods should just use those specs and now instantiate spec types.  Also any instrumentation examples need to be updated as well.

### Acceptance Criteria
Instrumentation using shim should instantiate the relevant spec type and not rely on the shim helpers to do this.  Another indicator is the instrumentation that uses shim should no longer return object literals with various keys.

### PR DESCRIPTION
**Pull request by svetlanabrennan**
## Description

Update specs from core to ioredis instrumentation. 

## Related Issues

Partially fixes #2022

### CODE REVIEW

File: lib/instrumentation/director.js [hunk 1]
```diff
@@ -14,36 +16,47 @@ module.exports = function initialize(agent, director, moduleName, shim) {
...
-        },
-        next: shim.LAST
-      })
-    }
-  })
+  shim.wrapMiddlewareMounter(
+    proto,
+    methods,
+    new MiddlewareMounterSpec({
+      // do I use MiddlewareMounterSpec here
```
Comment (svetlanabrennan) [inline lib/instrumentation/director.js:23]: @bizob2828 I use the MiddlewareMounterSpec here, correct?

File: lib/instrumentation/mongodb/common.js [hunk 2]
```diff
@@ -123,7 +123,7 @@ common.makeBulkDescFunc = function makeBulkDescFunc(shim) {
  *
  * @param {Shim} shim instance of shim
  * @param {object} instrumenter instance of mongo APM class
- * @param {object} [options={}] provide command names to skip updating host/port as they are unrelated to the active query.  This is only in v3 because after every command is runs `endSessions` which runs on the admin database
```
Comment (svetlanabrennan) [inline lib/instrumentation/mongodb/common.js:126]: Not sure why this change happened here. I tried to revert it but it's not working.

Comment (bizob2828) [reply lib/instrumentation/mongodb/common.js:126]: can't you just copy what was there previously and recommit in that case?

File: lib/instrumentation/director.js (hunk 1 above)
Comment (bizob2828) [reply lib/instrumentation/director.js:23]: yea this is correct

Comment (bizob2828) [reply lib/instrumentation/director.js:23]: I would remove that comment

File: lib/instrumentation/express.js [hunk 3]
```diff
@@ -88,19 +88,23 @@ function wrapExpress4(shim, express) {
     }
   })
```
Comment (bizob2828) [inline lib/instrumentation/express.js:89]: you're missing `recordRender` in this file

**Review by bizob2828 — CHANGES_REQUESTED** [review]
overall looks good, noted a few things that need updated

File: lib/instrumentation/mongodb/common.js (hunk 2 above)
Comment (svetlanabrennan) [reply lib/instrumentation/mongodb/common.js]: Just tried that and that fixed it. I did that before and it said no changes. Not sure what happened. But it's fixed now.

**Review by bizob2828 — APPROVED** [review]

**Review by jsumners-nr — APPROVED** [review]
